src/dropdown/dropdown.ts

```diff
@@ -1,7 +1,53 @@
-import {Directive, Input, Output, EventEmitter, ElementRef} from '@angular/core';
+import {
+  forwardRef,
+  Inject,
+  Directive,
+  Input,
+  Output,
+  EventEmitter,
+  ElementRef,
+  ContentChild,
+  OnChanges,
+  SimpleChanges
+} from '@angular/core';
 import {NgbDropdownConfig} from './dropdown-config';
 
 /**
+ */
+@Directive({selector: '[ngbDropdownMenu]', host: {'[class.dropdown-menu]': 'true', '[class.show]': 'isOpen'}})
+export class NgbDropdownMenu {
+  isOpen = false;
+
+  constructor(private _elementRef: ElementRef) {}
```

Comment:
Can't you inject the `NgbDropdown` as you do in the `NgbDropDownToggle` and remove internal `isOpen` state which is duplicated now?

Would it work if you just use `[class.show]: 'dropdown.isOpen()'`?
Can't see why you want to explicitly have `open` and `show` methods